hubcap/package.py

```diff
@@ -3,51 +3,160 @@
 import logging
 import os
 import yaml
+from typing import List, Dict, Any
 
 from git import Repo
 from pathlib import Path
 
 from hubcap import records
 from hubcap import version
-from hubcap.git_helper import clone_repo, run_cmd
+from hubcap.git_helper import clone_repo, run_cmd, GitOperationError
+
+
+class PackageError(Exception):
+    """Custom exception for package operation failures"""
+
+    pass
 
 
 def clone_package_repos(package_maintainer_index, path):
```

Comment:
Not a huge deal, but could add type hint of the return value as `List[str]` as we now return `failed_repos`